### PR TITLE
Flexible case tiles in case list

### PR DESCRIPTION
**Pull request by amstone326**
cross-request: https://github.com/dimagi/commcare-core/pull/363

I would NOT recommend reviewing this commit-by-commit, because I made a lot of incremental changes that didn't end up in the final version

### CODE REVIEW

File: app/src/org/commcare/views/EntityViewTile.java [hunk 1]
```diff
@@ -61,71 +62,67 @@
...
-
-    // constants used to calibrate how many tiles should be shown on a screen
+    private final int DEFAULT_TILE_PADDING_HORIZONTAL =
+            (int)getResources().getDimension(R.dimen.row_padding_horizontal);
+    private final int DEFAULT_TILE_PADDING_VERTICAL =
+            (int)getResources().getDimension(R.dimen.row_padding_vertical);
+
+    // Constants used to calibrate how many tiles should be shown on a screen at once -- This is
+    // saying that we expect a device with a density of 160 dpi and a height of 4 inches to look
+    // good in portrait mode with 4 tiles of 6 rows each being displayed at once.
```
Comment (amstone326) [inline app/src/org/commcare/views/EntityViewTile.java]: The preexisting defaults that were being used to calibrate the # of tiles shown on a screen were a bit wonky. 7 was very high for `DEFAULT_NUM_TILES_PER_SCREEN_PORTRAIT` (unless the default was meant to be on a tab? not sure). The preexisting strategy was also not taking absolute screen height into account at all, which was causing case tiles to appear very inconsistently on a phone vs. tablet, regardless of screen density. 

The comment here and the comment above `computeNumTilesPerScreen()` should make it clear what the new strategy I'm using is, but let me know if it's not.

**Comment by amstone326** [comment]
Screenshot of (essentially) what the UATBC calendar case list will look like, using this code and custom case tile xml:
uatbc_on_phone

**Comment by amstone326** [comment]
And screenshots of what all of the other (approved) usages of case tiles look like when rendered with this code, to confirm that they all still look correct:

ICDS on a high-density phone:
icds_on_phone

mLabour on an average-density tablet:
mlabour_on_tab

mLabour on a high-density phone: 
mlabour_on_phone

REC on an average-density tablet:
rec_on_tab

REC on a high-density phone:
rec_on_phone

**Comment by amstone326** [comment]
fyi @proteusvacuum

**Comment by czue** [comment]
this is awesome

**Comment by proteusvacuum** [comment]
😍

File: app/src/org/commcare/views/EntityViewTile.java [hunk 2]
```diff
@@ -251,10 +287,13 @@ public void addFieldViews(Context context, Detail detail, Entity entity) {
...
 
-        this.setBackgroundDrawable(null);
+        setPaddingIfNotInGridView();
 
-        this.setPadding(ROW_PADDING_HORIZONTAL, ROW_PADDING_VERTICAL,
-                ROW_PADDING_HORIZONTAL, ROW_PADDING_VERTICAL);
+        if (tileBeingShownInGridView()) {
+            // Fake dividers between each square in the grid view by using contrasting
+            // background colors for the grid view as a whole and each element in the grid view
+            setBackgroundColor(Color.parseColor("#ffffff"));
```
Comment (phillipm) [inline app/src/org/commcare/views/EntityViewTile.java]: `Color.parseColor("#ffffff")` can be `Color.WHITE`, right?

**Pull request merged**